sdk/keyvault/azure-security-keyvault-secrets/src/secret_serializers.cpp

```diff
@@ -111,3 +111,59 @@ void KeyVaultSecretSerializer::KeyVaultSecretDeserialize(
   JsonOptional::SetIfExists<std::string>(
       secret.Properties.ContentType, jsonParser, _detail::ContentTypePropertyName);
 }
+
+// serializes a set secret parameters object
+std::string KeyVaultSecretSerializer::KeyVaultSecretSerialize(KeyVaultSecret const& parameters)
+{
+  Azure::Core::Json::_internal::json payload;
+  using namespace Azure::Security::KeyVault::Secrets::_detail;
+
+  // value is required
+  payload[ValuePropertyName] = parameters.Value;
+
+  // all else is optional
+  JsonOptional::SetFromNullable(
+      parameters.Properties.ContentType, payload, ContentTypePropertyName);
+
+  Azure::Core::Json::_internal::json attributes;
+
+  JsonOptional::SetFromNullable<Azure::DateTime, int64_t>(
+      parameters.Properties.CreatedOn,
+      attributes,
+      CreatedPropertyName,
+      PosixTimeConverter::DateTimeToPosixTime);
+  JsonOptional::SetFromNullable(parameters.Properties.Enabled, attributes, EnabledPropertyName);
+  JsonOptional::SetFromNullable<Azure::DateTime, int64_t>(
+      parameters.Properties.ExpiresOn,
+      attributes,
+      ExpPropertyName,
+      PosixTimeConverter::DateTimeToPosixTime);
+  JsonOptional::SetFromNullable<Azure::DateTime, int64_t>(
+      parameters.Properties.NotBefore,
+      attributes,
+      NbfPropertyName,
+      PosixTimeConverter::DateTimeToPosixTime);
+  JsonOptional::SetFromNullable(
+      parameters.Properties.RecoverableDays, attributes, RecoverableDaysPropertyName);
+  JsonOptional::SetFromNullable(
+      parameters.Properties.RecoveryLevel, attributes, RecoveryLevelPropertyName);
+  JsonOptional::SetFromNullable<Azure::DateTime, int64_t>(
+      parameters.Properties.UpdatedOn,
+      attributes,
+      UpdatedPropertyName,
+      PosixTimeConverter::DateTimeToPosixTime);
+
+  // optional tags
+  Azure::Core::Json::_internal::json tags;
+
+  for (auto iterator : parameters.Properties.Tags)
+  {
+    tags[iterator.first] = iterator.second;
+  }
+
+  attributes[TagsPropertyName] = tags;
```

Comment:
Just found out that for std containers using std types (like string), we can create the json node directly like:

```suggestion
  attributes[TagsPropertyName] =   Azure::Core::Json::_internal::json(parameters.Properties.Tags);
```

This help us to remove the loop.  json type can take any std container using std types as T and V